pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -580,7 +580,7 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 		// volume.volumeSpecName is actually InnerVolumeSpecName. It will not be used
 		// for volume cleanup.
 		// TODO: in case pod is added back before reconciler starts to unmount, we can update this field from desired state information
-		outerVolumeSpecName: volume.volumeSpecName,
+		outerVolumeSpecName: volumeSpec.Name(),
```

Comment:
I think outerVolumeSpecName is basically not possible to reconstruct. It is a name defined in pod spec, iirc.
So volumeSpec.Name() will not return the correct one either. 
We are not using this outerVolumeSpecName actually, I think.

Reply:
Many thanks for your comments. As described in issue #102402, actually outerVolumeSpecName is used when checking if all volumes are mounted. If innerName and outerName of an ephemeral volume are different, it will never be seen as mounted.

```
func (vm *volumeManager) getUnmountedVolumes(podName types.UniquePodName, expectedVolumes []string) []string {
	mountedVolumes := sets.NewString()
	for _, mountedVolume := range vm.actualStateOfWorld.GetMountedVolumesForPod(podName) {
		mountedVolumes.Insert(mountedVolume.OuterVolumeSpecName)
	}
	return filterUnmountedVolumes(mountedVolumes, expectedVolumes)
}
```

And yes, volumeSpec.Name() is not outerVolumeName actually, I just do an internal convension from outerName to innerName in `ConstructVolumeSpec` implemenation in our self-developed plugin, which may not implemented by other third-party storage plugins.